src/openfermion/ops/_code_operator.py

```diff
@@ -0,0 +1,298 @@
+import copy
+
+import numpy
+import scipy
+import scipy.sparse
+
+from openfermion.ops import SymbolicBinary
+
+
+def _shift_decoder(decode, shift_constant):
```

Comment:
Python does not really have a concept of private functions. I suppose this underscore is meant to indicate that it is an internal helper function. We've allowed that so far for methods on classes but not for external functions. I feel that this should either be a method on the class or you should just make it a public function (get rid of the underscore).